packages/connect-node/src/private/web-header-to-node-headers.ts

```diff
@@ -67,3 +77,47 @@ export function webHeaderToNodeHeaders(
   }
   return o;
 }
+
+export function parseGrpcTimeoutFromHeader(value: string): number | null {
```

Comment:
Can you move this function to connect-core? Best to prefix the name with "grpc", as in "grpcParseTimeout". 

It will be helpful to accept `string | null` as an argument, similar to `grpcParseContentType()` - this lets us pass a value from a `Headers` object without extra if-statements.

The errors are all correct, but I think it might be easier to work with if we didn't raise an error here, and instead _returned_ an error. We have three possible return values if we accept `string | null` as input:
1. `number` - parsed timeout in milliseconds
2. `undefined` - no input was given
3. `?` - input is invalid

So this could be `function grpcParseTimeout(val: string|null): number | undefined | ConnectError`, used as:

```ts
const timeout = grpcParseTimeout(requestHeader.get("Grpc-Timeout"));
if (typeof timeout == "number") {
  res.setTimeout(...);
}

// validate request content type

if (timeout instanceof ConnectError) {
  return void endWithGrpcWebTrailer(
    res,
    context,
    timeout
  );
}
```

Am open to alternatives here, but I think we should avoid raising errors because it complicates control flow.

Reply:
thats a great idea and definitely in favor of the protocol returning the error instead of raising here in the check